common/src/main/java/com/wynntils/core/features/FeatureRegistry.java

```diff
@@ -4,66 +4,127 @@
+import com.wynntils.features.user.DialogueOptionOverrideFeature;
+import com.wynntils.features.user.EmeraldPouchHotkeyFeature;
+import com.wynntils.features.user.GammabrightFeature;
+import com.wynntils.features.user.HealthPotionBlockerFeature;
+import com.wynntils.features.user.IngredientPouchHotkeyFeature;
+import com.wynntils.features.user.ItemGuessFeature;
+import com.wynntils.features.user.ItemHighlightFeature;
+import com.wynntils.features.user.ItemScreenshotFeature;
+import com.wynntils.features.user.ItemStatInfoFeature;
+import com.wynntils.features.user.MountHorseHotkeyFeature;
+import com.wynntils.features.user.MythicBlockerFeature;
+import com.wynntils.features.user.PlayerGhostTransparencyFeature;
+import com.wynntils.features.user.SoulPointTimerFeature;
+import com.wynntils.features.user.WynncraftButtonFeature;
 import com.wynntils.mc.event.ClientTickEvent;
 import com.wynntils.mc.event.RenderEvent;
 import com.wynntils.mc.utils.CrashReportManager;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wc.utils.WynnUtils;
-import java.util.HashMap;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 /** Loads {@link Feature}s */
 public class FeatureRegistry {
-    private static final Map<Class<? extends Feature>, Feature> FEATURES = new HashMap<>();
-    private static final Map<Class<? extends Overlay>, Overlay> OVERLAYS = new HashMap<>();
+    private static final List<Feature> FEATURES = new ArrayList<>();
+    private static final List<Overlay> OVERLAYS = new ArrayList<>();
 
     public static void registerFeature(Feature feature) {
         if (feature instanceof Overlay overlay) {
-            OVERLAYS.put(overlay.getClass(), overlay);
+            OVERLAYS.add(overlay);
         }
 
-        FEATURES.put(feature.getClass(), feature);
+        FEATURES.add(feature);
+
+        initializeFeature(feature);
+    }
+
+    private static void initializeFeature(Feature feature) {
+        Class<? extends Feature> featureClass = feature.getClass();
+
+        // instance field
+        try {
+            Field instanceField = featureClass.getDeclaredField("INSTANCE");
+            instanceField.set(null, feature);
+        } catch (Exception e) {
+            Reference.LOGGER.error("Failed to create instance object in " + featureClass.getName());
+            e.printStackTrace();
+            return;
+        }
+
+        // flag as event listener
+        if (featureClass.isAnnotationPresent(EventListener.class)) {
+            feature.setupEventListener();
+        }
+
+        // register key binds
+        if (featureClass.isAnnotationPresent(KeyBinds.class)) {
+            for (Field f : featureClass.getDeclaredFields()) {
+                if (!f.getType().equals(KeyHolder.class)) continue;
+
+                try {
+                    f.setAccessible(true);
```

Comment:
It might be better to have a full reflection utils method which also handles final modifier